src/plugins/intel_cpu/src/utils/shape_inference/shape_inference.cpp

```diff
@@ -503,6 +504,8 @@ std::shared_ptr<IShapeInfer> make_shape_inference(const std::shared_ptr<ngraph::
         return make_shared_entryIOC(node);
     } else if (auto node = ov::as_type_ptr<ov::opset1::GatherTree>(op)) {
         return make_shared_entryIO(node);
+    } else if (auto node = ov::as_type_ptr<ov::opset5::GRUSequence>(op)) {
+        return make_shared_entryIO(node);
```

Comment:
@jane-intel @nshchego Should GRUSequence `shape_infer` be enabled for cpu here?

As I can see in PR https://github.com/openvinotoolkit/openvino/pull/9002, `batch` dynamism for rnn ops was resolved in other way, and there is a restriction in the cpu implementation that W, R, B inputs are required to be Constant anyway: https://github.com/openvinotoolkit/openvino/blob/3c24ee6cdaee75176f309dff6ce57af5aa4e8110/src/plugins/intel_cpu/src/nodes/rnn.cpp#L202-L208

Reply:
As I understand it, we should not add an instantiation here in case of internal dynamism, i.e. when the output shape depend on input data.
@jane-intel ,  could you confirm that?

Reply:
Just to clarify, `GRUSequence` output shape doesn't depend on the inputs data, but on the inputs shapes (and op attributes).

Reply:
We need to include it here. When dynamic input shape would be supported -- CPU plugin would use this code to infer shape with or without restrictions on constant weights inputs -- it doesn't really matter from shape inference stand point.